federation/pkg/federation-controller/util/deletionhelper/deletion_helper.go

```diff
@@ -104,37 +100,27 @@ func (dh *DeletionHelper) EnsureFinalizers(obj runtime.Object) (
 }
 
 // Deletes the resources corresponding to the given federated resource from
-// all underlying clusters, unless it has the FinalizerOrphan finalizer.
-// Removes FinalizerOrphan and FinalizerDeleteFromUnderlyingClusters finalizers
-// when done.
+// all underlying clusters if the resource has the FinalizerDeleteFromUnderlyingClusters
+// finalizer and does not have the FinalizerOrphan finalizer.
 // Callers are expected to keep calling this (with appropriate backoff) until
 // it succeeds.
-func (dh *DeletionHelper) HandleObjectInUnderlyingClusters(obj runtime.Object) (
-	runtime.Object, error) {
+func (dh *DeletionHelper) HandleObjectInUnderlyingClusters(obj runtime.Object) error {
 	objName := dh.objNameFunc(obj)
 	glog.V(2).Infof("Handling deletion of federated dependents for object: %s", objName)
 	if !dh.hasFinalizerFunc(obj, FinalizerDeleteFromUnderlyingClusters) {
```

Comment:
If the resource doesn't have this finalizer, but does have the orphan finalizer, should this method be removing the orphan finalizer to ensure object removal?  Or is this finalizer being absent indicate that the ophan finalizer will also be absent?

cc: @nikhiljindal